src/main/java/com/globalitgeeks/examninja/service/ExamResultService.java

```diff
@@ -0,0 +1,152 @@
+
+package com.globalitgeeks.examninja.service;
+
+
+import com.globalitgeeks.examninja.dto.ExamResultResponse;
+import com.globalitgeeks.examninja.dto.ExamSubmissionRequest;
+import com.globalitgeeks.examninja.exception.ExamDataBaseOperationException;
+import com.globalitgeeks.examninja.exception.InvalidExamDataException;
+import com.globalitgeeks.examninja.exception.QuestionNotFoundException;
+import com.globalitgeeks.examninja.model.ExamResult;
+import com.globalitgeeks.examninja.model.ExamResultDetail;
+import com.globalitgeeks.examninja.model.Question;
+import com.globalitgeeks.examninja.model.User;
+import com.globalitgeeks.examninja.repository.ExamResultDetailRepository;
+import com.globalitgeeks.examninja.repository.ExamResultRepository;
+import com.globalitgeeks.examninja.repository.QuestionRepository;
+import com.globalitgeeks.examninja.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class ExamResultService {
+    @Autowired
+    private AnswerService answerService;
+
+    @Autowired
+    private QuestionRepository questionRepository;
+
+    @Autowired
+    private ExamResultDetailRepository examResultDetailRepository;
+
+    @Autowired
+    private ExamResultRepository examResultRepository;
+
+   public ExamResultResponse processSubmittedTest(ExamSubmissionRequest request) {
+       Long testId = request.getTestId();
+       Long userId = request.getId();
+
+       // Combine submissionDate and submissionTime from request
+       LocalDate submissionDate = request.getSubmissionDate();
+       LocalTime submissionTime = request.getSubmissionTime();
+       LocalDateTime submissionDateTime = LocalDateTime.of(submissionDate, submissionTime);
+
+       // Fetch all answers from AnswerService
+       Map<String, String> allAnswers = answerService.getAllAnswers();
+       if (allAnswers.isEmpty()) {
+           throw new InvalidExamDataException("No answers found for the user.");
+       }
+
+       // Filter answers for the specific student and test
+       Map<Long, String> studentTestAnswers = filterStudentTestAnswers(allAnswers, userId, testId);
+
+       // Process answers and calculate the result
+       int totalQuestions = studentTestAnswers.size();
+       int correctAnswers = 0;
+
+       if (totalQuestions == 0) {
+           // If there are no answers, return a score of 0 and fail status
+           return new ExamResultResponse(testId, userId, 0, 65, "FAIL");
```

Comment:
65 should be captured in a final constant such as private static final PASS_MARKS_CUT_OFF = 65; to make the code more readable.